commons/src/main/java/org/infinispan/commons/api/functional/FunctionalMap.java

```diff
@@ -0,0 +1,634 @@
+package org.infinispan.commons.api.functional;
```

Comment:
What does putting this in commons buy for us?  It seems we don't get a whole lot and we have to start pulling down core deps like Status.